### ISSUE TITLE
ArgumentTypeCoercion false positive with static

### ISSUE THREAD

**Issue by enumag** [vimeo/psalm#4157]
https://psalm.dev/r/fa3dc15c96

**Comment by weirdan** [comment]
I don't think it's a bug. E.g. if there's class `C` extending `B`, you would be able to call `x` like this:
```php
x(new C, new B);
```
which is probably not something you want here.

**Comment by muglug** [comment]
Yup

**Comment by enumag** [comment]
Ah you're right. The issue disappears when I add `final`. Thanks!
